python/pysmurf/client/util/smurf_util.py

```diff
@@ -835,11 +838,26 @@ def stream_data_on(self, write_config=False, data_filename=None,
             return data_filename
 
 
-    def stream_data_off(self, write_log=True):
+    def stream_data_off(self, write_log=True, register_file=False):
         """
         Turns off streaming data.
+
+        Args:
+            write_log (bool):
+                Whether to log the CA commands or not. Default False
+            register_file (bool):
+                If true, the stream data file will be registered through the
+                publisher.
         """
+        if register_file:
```

Comment:
I think can refactor this section of the code to be something like:
```
self.close_data_file(write_log=write_log)

if register_file:
    datafile = self.get_data_file_name().tostring().decode()
    self.log("Registering File {}".format(datafile))
    self.pub.register_file(datafile, 'data', format='dat')
```
`get_data_file_name()` will return the name of the data file, even after the file is closed.